### PR TITLE
Create JanDan.xml

### PR DESCRIPTION
**Pull request by renyouguo**

### CODE REVIEW

**Comment by J0WI** [comment]
I think you can redirect cdn and www to ^.
I also found a mismatch on `g.jandan.net`

File: src/chrome/content/rules/JianDan.xml [hunk 1]
```diff
@@ -1,29 +1,35 @@
...
-	  <test url="http://jandan.net/2016/06/03/fucking-kids.html" />
-	  <test url="http://jandan.net/ofk.gif" />
-	  <test url="http://jandan.net/share/2016/06/qr-79613.png" />
+	<target host="www.jiandan.net" />
+		<test url="http://jandan.net/ofk.gif" />
+		<test url="http://jandan.net/share/2016/06/qr-79613.png" />
+		<test url="http://www.jandan.net/ofk.gif" />
+		<test url="http://www.jandan.net/share/2016/06/qr-79613.png" />
+	<exclusion pattern="^http://(www\.)?jiandan\.net/(?![\w/-]+\.(gif|png))" />
+	<rule from="^http://(www\.)?jiandan\.net/(.+)" to="https://jiandan.net/$1" />
```
Comment (J0WI) [inline src/chrome/content/rules/JianDan.xml]: The from `(.+)` to `$1` is not necessary.

File: src/chrome/content/rules/JianDan.xml [hunk 2]
```diff
@@ -1,29 +1,35 @@
...
+
+	<target host="cdn.jiandan.net" />
+		<test url="http://cdn.jandan.net/static/img/favicon.ico" />
+		<test url="http://cdn.jandan.net/static/js/common.js" />
+		<test url="http://cdn.jandan.net/wp-content/themes/egg/style.css" />
+	<rule from="^http://cdn\.jiandan\.net/(.+)" to="https://jiandan.net/$1" />
```
Comment (J0WI) [inline src/chrome/content/rules/JianDan.xml]: same here

File: src/chrome/content/rules/JianDan.xml [hunk 3]
```diff
@@ -1,29 +1,35 @@
...
 
 	<target host="tankr.net" />
```
Comment (J0WI) [inline src/chrome/content/rules/JianDan.xml]: How is this host related to jandan.net?

**Comment by J0WI** [comment]
This needs some more tests for all the negative lookahead expressions

File: src/chrome/content/rules/JianDan.xml (hunk 3 above)
Comment (renyouguo) [inline src/chrome/content/rules/JianDan.xml]: @J0WI Tankr.net仅供煎蛋小编内部使用。
English: Tankr.net is only used by jiandan teamers.

File: src/chrome/content/rules/JianDan.xml (hunk 1 above)
Comment (renyouguo) [inline src/chrome/content/rules/JianDan.xml]: @J0WI www.jidan.net/$ has MCB, so I think it's necessary.

Comment (J0WI) [inline src/chrome/content/rules/JianDan.xml]: I would then add an exclusion for `/$`

File: src/chrome/content/rules/JianDan.xml (hunk 3 above)
Comment (J0WI) [inline src/chrome/content/rules/JianDan.xml]: There are also:
www.tankr.net
jdstatic.tankr.net

Comment (J0WI) [inline src/chrome/content/rules/JianDan.xml]: Also not that there is different content served on https than on http (not sure if it matters).

Comment (renyouguo) [inline src/chrome/content/rules/JianDan.xml]: @J0WI You are right. May I delete it or add one new rule?

Comment (J0WI) [inline src/chrome/content/rules/JianDan.xml]: You can just delete it

**Comment by renyouguo** [comment]
@J0WI Stupid error has been fixed.

File: src/chrome/content/rules/JanDan.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		<test url="http://jandan.net/ofk.gif" />
+		<test url="http://jandan.net/wp-post-views.php" />
+		<test url="http://jandan.net/share/2016/06/qr-80305.png" />
+		<test url="http://jandan.net/share/2016/06/qr-79613.png" />
+		<test url="http://www.jandan.net/ofk.gif" />
+		<test url="http://www.jandan.net/wp-post-views.php" />
+		<test url="http://www.jandan.net/share/2016/06/qr-80305.png" />
+		<test url="http://www.jandan.net/share/2016/06/qr-79613.png" />
+
+	<!--
```
Comment (J0WI) [inline src/chrome/content/rules/JanDan.xml]: Not sure why you excluded this now?

Comment (renyouguo) [inline src/chrome/content/rules/JanDan.xml]: I got it mismatch now.

**Comment by J0WI** [comment]
It seems that they updated their site, I now get content served from `sslcdn.jandan.net`.
Could you recheck your rules and the mixed content issues?

**Comment by J0WI** [comment]
Thanks, that simplified a lot.
